### PR TITLE
Viewer material variants

### PR DESCRIPTION
**Pull request by ryantrem**
This PR adds support for material variants along with default UI in the new Viewer.

https://github.com/user-attachments/assets/bf657a97-8c98-4077-ab01-78fd6bae5dd2

- Updates the `KHR_materials_variants` extension to have a callback option that "returns" a "controller" (available variants + selected variant). `KHR_materials_variants` already has both instance and static functions for getting material variants and selecting a material variant, but this would require importing the extension, and we don't want the extension included in the bundle statically (we only want to pay the bundle cost of the extension when it is actually used). This "controller callback" is kind of a new pattern for glTF extensions using the newish options support. Another way to do this would be to have a higher level callback option for when a specific loader plugin is created (analogous to `OnPluginActivatedObservable`) and when a specific glTF extension is created (analogous to `onExtensionLoadedObservable`) and use the extension instance directly. I don't really like this because to me it seems like a complex API (in the same way the static observables are complex and often a sticking point for users). Another option yet would be to have some kind of extensible output object in the same way options are an extensible input object. For example, `AssetContainer` could have an `extras` property (or something like that) where the underlying `AssetExtras` type could be extended by loader plugins and glTF extensions. Then you might do something like (`const variants = assetContainer.extras.materialVariants` and `assetContainer.extras.selectedMaterialVariant = someVariant`). That would be a bigger change though, and I'm not sure how we would make something like this work across the different loader functions. I think it could make sense for `loadAssetContainerAsync`, but maybe not the other loader functions. Thoughts @bghgary?
- Viewer uses the "controller" returned by `KHR_materials_variants` to expose a simple material variants API.
- Viewer element exposes material variants and also provides default UI (variant selection).
- A bit of refactoring in the viewer element to make it easy to have any combination of toolbar controls (animation, material variants, hotspots) and put a vertical separator between each one.

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Review by bghgary — APPROVED** [review]
Discussed offline

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15920/merge

Test environment: 
https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15920/merge/index.html

To test a playground add it to the URL, for example:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15920/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15920/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15920/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15920/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15920/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15920/merge#BCU1XR#0

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15920/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15920/merge/testResults/webgpuplaywright/index.html

File: packages/dev/loaders/src/glTF/2.0/Extensions/KHR_materials_variants.ts [hunk 1]
```diff
@@ -190,6 +210,31 @@ export class KHR_materials_variants implements IGLTFLoaderExtension {
...
+    public onReady(): void {
+        const rootNode = this._loader.rootBabylonMesh;
+        if (rootNode) {
+            this._loader.parent.extensionOptions[NAME]?.onLoaded?.({
+                get variants() {
+                    return KHR_materials_variants.GetAvailableVariants(rootNode);
+                },
+                get selectedVariant(): string {
+                    const lastSelectedVariant = KHR_materials_variants.GetLastSelectedVariant(rootNode);
+                    if (!lastSelectedVariant) {
```
[the remark this answers is not in the document]
Comment (bghgary) [reply packages/dev/loaders/src/glTF/2.0/Extensions/KHR_materials_variants.ts:223]: I don't think so. Not everyone will want to do this.

**Pull request merged**